packages/DataGrid/CsvGenerator.ts

```diff
@@ -2,17 +2,40 @@ import { Downloader } from '@3mo/downloader'
 import type { DataGrid } from './DataGrid.js'
 
 export class CsvGenerator {
-	static generate<TData>(dataGrid: DataGrid<TData, any>) {
-		const flattenedData = [...dataGrid['getFlattenedData']()]
+	static fetchAll = async <TData>(dataGrid: any): Promise<Array<TData>> => {
```

Comment:
@a11delavar, I do not really like the current solution but do not have any better ideas.